integration-test/mqtt/mqtt_system_test.c

```diff
@@ -392,15 +419,28 @@ static void establishMqttSession( MQTTContext_t * pContext,
 
     if( useLWTClientIdentifier )
     {
-        connectInfo.pClientIdentifier = TEST_CLIENT_IDENTIFIER_LWT;
-        connectInfo.clientIdentifierLength = TEST_CLIENT_IDENTIFIER_LWT_LENGTH;
+        /* Populate client identifier for connection with LWT topic with random number. */
+        connectInfo.clientIdentifierLength =
+            snprintf( clientIdBuffer, sizeof( clientIdBuffer ),
+                      "%d%s",
+                      clientIdRandNumber,
+                      TEST_CLIENT_IDENTIFIER_LWT );
+        connectInfo.pClientIdentifier = clientIdBuffer;
     }
     else
     {
-        connectInfo.pClientIdentifier = TEST_CLIENT_IDENTIFIER;
-        connectInfo.clientIdentifierLength = TEST_CLIENT_IDENTIFIER_LENGTH;
+        /* Populate client identifier with random number. */
+        connectInfo.clientIdentifierLength =
+            snprintf( clientIdBuffer,
+                      sizeof( clientIdBuffer ),
+                      "%d%s", clientIdRandNumber,
```

Comment:
Why not `%03d` so all client IDs will be the same width? Then the length will always be TEST_CLIENT_IDENTIFIER_LENGTH + 3

Reply:
What's the exact advantage? `snprintf` does return the length..

Reply:
None, I just thought it'd be cleaner if they were always the same size